ocaml/typing/jkind.mli

```diff
@@ -41,267 +41,417 @@
 (* CR externals: Move to mode.ml. But see
    https://github.com/goldfirere/flambda-backend/commit/d802597fbdaaa850e1ed9209a1305c5dcdf71e17
    first, which was reisenberg's attempt to do so. *)
-module Externality : sig
-  type t = Jkind_types.Externality.t =
-    | External (* not managed by the garbage collector *)
-    | External64 (* not managed by the garbage collector on 64-bit systems *)
-    | Internal (* managed by the garbage collector *)
 
-  val le : t -> t -> bool
+exception Unexpected_higher_jkind of string
```

Comment:
Does this really need to be exported? That makes me a little unhappy about the interface here. Exporting this subtly suggests that any arbitrary function in this module might throw this exception.

Reply:
That's true, it doesn't have to be. It should only be raised by `to_type_jkind` in practice, and is not expected to be caught. I could be using `assert false` but I used this to make bugs more obvious to track down. Also, exporting right now is entirely unnecessary.